tc/tc_util.h

```diff
@@ -24,14 +24,14 @@ struct qdisc_util {
 	struct  qdisc_util *next;
 	const char *id;
 	int (*parse_qopt)(struct qdisc_util *qu, int argc,
-			  char **argv, struct nlmsghdr *n);
+			  char **argv, struct nlmsghdr *n, char *dev);
 	int (*print_qopt)(struct qdisc_util *qu,
 			  FILE *f, struct rtattr *opt);
 	int (*print_xstats)(struct qdisc_util *qu,
 			    FILE *f, struct rtattr *xstats);
 
 	int (*parse_copt)(struct qdisc_util *qu, int argc,
-			  char **argv, struct nlmsghdr *n);
+			  char **argv, struct nlmsghdr *n, char *dev);
```

Comment:
Instead of changing the interface of `parse_copt()`, which requires changing the function signatures in many files, how about we put the device string into a new field in `struct qdisc_util` (instead of as a local variable in `tc_qdisc_modify()` and `tc_class_modify()`).

That way, this information can be passed to all of the individual qdiscs and classes in the same way that the ID field is inside of `struct qdisc_utils`.

Reply:
OK, I think you can keep the code the way you had it (by changing the function signature) for now.